libs/langchain/langchain/retrievers/document_compressors/rankllm_rerank.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence
+
+from langchain_core.documents import Document
+from langchain_core.pydantic_v1 import Extra, Field, PrivateAttr, root_validator
+
+from langchain.callbacks.manager import Callbacks
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+
+if TYPE_CHECKING:
+    from rank_llm.data import Candidate, Query, Request
+else:
+    # Avoid pydantic annotation issues when actually instantiating
+    # while keeping this import optional
+    try:
+        from rank_llm.data import Candidate, Query, Request
+    except ImportError:
+        pass
+
+
+class RankLLMRerank(BaseDocumentCompressor):
+    """Document compressor using Flashrank interface."""
+
+    client: Any = None
+    """RankLLM client to use for compressing documents"""
+    top_n: int = Field(default=3)
+    """Top N documents to return."""
+    model: str = Field(default="zephyr")
+    """Name of model to use for reranking."""
+    step_size: int = Field(default=10)
+    """Step size for moving sliding window."""
+    gpt_model: str = Field(default="gpt-3.5-turbo")
+    """OpenAI model name."""
+    _retriever: Any = PrivateAttr()
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    @root_validator(pre=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate python package exists in environment."""
+
+        if not values.get("client"):
+            client_name = values.get("model", "zephyr")
+
+            try:
+                model_enum = ModelType(client_name.lower())
+            except ValueError:
+                raise ValueError(
+                    "Unsupported model type. Please use 'vicuna', 'zephyr', or 'gpt'."
+                )
+
+            try:
+                if model_enum == ModelType.VICUNA:
+                    from rank_llm.rerank.vicuna_reranker import VicunaReranker
+
+                    values["client"] = VicunaReranker()
+                elif model_enum == ModelType.ZEPHYR:
+                    from rank_llm.rerank.zephyr_reranker import ZephyrReranker
+
+                    values["client"] = ZephyrReranker()
+                elif model_enum == ModelType.GPT:
+                    from rank_llm.rerank.rank_gpt import SafeOpenai
+                    from rank_llm.rerank.reranker import Reranker
+
+                    from langchain.llms.openai import OpenAI
+
+                    llm = OpenAI(
```

Comment:
you are importing just to access the api key? lets not do that, surely there has to be a more direct way